### PR TITLE
Improvements to the one variable

### PR DESCRIPTION
**Pull request by Ivanluv**
This is from PR #6004

### CODE REVIEW

File: instat/dlgOneVarFitModel.vb [hunk 1]
```diff
@@ -14,720 +14,925 @@
...
+        If rdoGeneralCase.Checked Then
+            ucrSaveModel.SetCheckBoxText("Save Model")
+            ucrSaveModel.SetPrefix("dist")
+        ElseIf rdoTest.Checked Then
+            ucrSaveModel.SetCheckBoxText("Save Test")
+            ucrSaveModel.SetPrefix("test")
+        ElseIf rdoEstimate.Checked Then
+            ucrSaveModel.SetCheckBoxText("Save Estimate")
+            ucrSaveModel.SetPrefix("ci")
+        End If
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:642]: Add an `Else` with a developer error?
There's an example in `dlgCalculator.vb`:

`MsgBox("Developer error: there is no Calculator keyboard called" & Chr(34) & strNewDefaultKeyboard & Chr(34) & vbNewLine & "Default keyboard will be selected.")`

File: instat/dlgOneVarFitModel.vb [hunk 2]
```diff
@@ -14,720 +14,925 @@
...
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsMedianCIFunction)
+                Case "normal"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsNormCIFunction)
+                Case "quantile"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsQuantileCIFunction)
+                Case "sd"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsSdCIFunction)
+                Case "variance"
+                    ucrBase.clsRsyntax.SetBaseRFunction(clsVarCIFunction)
+            End Select
```
Comment (lloyddewit) [inline instat/dlgOneVarFitModel.vb:885]: Add `Else` with developer error?

**Review by lloyddewit — COMMENTED** [review]
@Ivanluv thanks for the updates, there's just a couple of open questions about developer errors

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern you already tested this new functionality under PR #6004. There are relatively small changes since then. Please could you retest and approve? thanks

**Comment by rdstern** [comment]
@Ivanluv sorry - this is a complicated dialogue.  I am hoping to get it merged, though we will continue to find combinations that don't work.  But this one could usefully be fixed now.

I used the standard survey data set and the default test - which is binomial and variety - which is a factor.  It then gives the levels of the factor.  But this command also works for numeric variables.  So the levels control should remain open and the successive values become the levels.  You don't need to, but may wish to do an as-factor behind the scenes - perhaps Shadrack can help.  Currently it doesn't work well as you see here.

When I changed to fert from the variety it still kept the success="NEW" from the variety - so had zero.
When I tried again it allowed and correctly had zero - lowest level - which is great.  But I can't change the "level".

```
# Code generated by the dialog, One Variable Fit Model
fert <- data_book$get_columns_from_data(data_name="survey", col_names="fert")
last_model <- mosaic::binom.test(x=fert, p=0.5, conf.level=0.95, ci.method="Clopper-Pearson", success="NEW")
data_book$add_model(model_name="last_model", model=last_model, data_name="survey")
data_book$get_models(data_name="survey", model_name="last_model")

data:  fert  [with success = NEW]
number of successes = 0, number of trials = 36, p-value = 3e-11
alternative hypothesis: true probability of success is not equal to 0.5
95 percent confidence interval:
 0.00000 0.09739
sample estimates:
probability of success 
                     0 

rm(list=c("last_model", "fert"))

# Code generated by the dialog, One Variable Fit Model
fert <- data_book$get_columns_from_data(data_name="survey", col_names="fert")
last_model <- mosaic::binom.test(x=fert, p=0.5, conf.level=0.95, ci.method="Clopper-Pearson")
data_book$add_model(model_name="last_model", model=last_model, data_name="survey")
data_book$get_models(data_name="survey", model_name="last_model")

data:  fert  [with success = 0]
number of successes = 9, number of trials = 36, p-value = 0.004
alternative hypothesis: true probability of success is not equal to 0.5
95 percent confidence interval:
 0.1212 0.4220
sample estimates:
probability of success 
                  0.25 
```

**Comment by Ivanluv** [comment]
Am still working on having the levels for numeric columns

**Comment by Ivanluv** [comment]
@rdstern could you look at this again

**Review by rdstern — DISMISSED** [review]
@Ivanluv I am approving this, now because it works fine - at least the bit I checked, with binomial.
However, it does have a small problem to me in that the fert - that is a variable that could be either numeric or a factor is silently converted into a factor for the binomial - and then it remains as a factor afterwards.  I assume you need to make it a factor for your code?  I tried with fertgrp as character and it leaves it as character.  
I then tried to analyse fert with a t-test and it gives an error, because fert is now a factor.  
So, unless you need it to become a factor, then it should be left as numeric.  If it does need, for your code to become a factor, then ideally you return it to its previous type later, or you make a comment in the output window that the variable is now converted to factor.

**Review by rdstern — DISMISSED** [review]
@lloyddewit this is now working as far as I can see.  Great if it can be accepted

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Ivanluv Thanks, this looks good. I only had a few small suggestions

File: instat/dlgOneVarFitModel.vb [hunk 3]
```diff
@@ -14,720 +14,972 @@
...
+        Next
+
+        'converting back to Numeric 
+        If bTypeIsNumeric Then
+            clsConvertToColumnTypeFunction.AddParameter("to_type", Chr(34) & "numeric" & Chr(34), iPosition:=1)
+            frmMain.clsRLink.RunInternalScript(clsConvertToColumnTypeFunction.ToScript(), bSilent:=True)
+        End If
+
+        ucrInputSuccess.SetItems(lstFactor.ToArray)
+        ucrInputSuccess.SetText(lstFactor(0))
```
[the remark this answers is not in the document]
Comment (Ivanluv) [reply instat/dlgOneVarFitModel.vb:949]: I have added a developer error, Stating there are no factor levels to be displayed

File: instat/dlgOneVarFitModel.vb [hunk 4]
```diff
@@ -14,720 +14,972 @@
...
+        clsProportionFunction.RemoveParameterByName("success")
+    End Sub
+
+    Private Sub ucrInputTests_ControlValueChanged(ucrChangedControl As ucrCore) Handles ucrInputComboTests.ControlValueChanged, ucrInputComboEstimate.ControlValueChanged
+        SetTestEstimateBaseFunction()
+        ucrSaveModel.Reset()
+        AddAsNumeric()
+        EstimatesAsNumeric()
+        AddFactorLevels()
+        If ucrInputComboTests.GetText = "---------------------" Then
```
[the remark this answers is not in the document]
Comment (Ivanluv) [reply instat/dlgOneVarFitModel.vb:968]: this was to be used as a separator for tests within the Test combo box. I don't know is a constant will do the same

[the remark this answers is not in the document]
Comment (lloyddewit) [reply instat/dlgOneVarFitModel.vb:967]: Could you declare a constant (e.g. `Private ReadOnly strSeparator As String = "---------------------"`) and then use this constant here, and on line 95?

[the remark this answers is not in the document]
Comment (Ivanluv) [reply instat/dlgOneVarFitModel.vb]: Done

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern there are small code changes since your last approval. These should not have changed the functionality (in theory!). You may still want to do a small sanity check before approving again :) thanks

**Comment by shadrackkibet** [comment]
@lloyddewit I see this PR has two approvals already. Is there something else needed? or you just forget to merge?

**Pull request merged**